Automattic/wp-calypso
# Framework: Replace React.addons.PureRenderMixin with react-pure-render

This pull request seeks to replace `React.addons.PureRenderMixin` with the `react-pure-render` package. React addons were moved to separate packages in [React 0.14](https://facebook.github.io/react/blog/2015/10/07/react-v0.14.html) and currently cause warnings to appear in development environments. Facebook offers its own [`react-addons-pure-render-mixin` package](https://www.npmjs.com/package/react-addons-pure-render-mixin), but I suggest here that we use the [`react-pure-render` package](https://github.com/gaearon/react-pure-render) instead, as it includes a few useful variations for usage (i.e. function for use in React component classes).

__Testing instructions:__

- Verify that you can navigate the application without errors.

__Code verification:__

- Verify that `/addons` suffix in React import accurately reflects current existence of addon usage.
- Ensure that no usage of `React.addons.PureRenderMixin` remains

__Next steps:__

We should replace `shallowequal` usage with `react-pure-render`s own `react-pure-render/shallowEqual` module (see #1463, cc @ockham )
